modules/docker/host_test.go

```diff
@@ -0,0 +1,66 @@
+package docker
+
+import (
+	"fmt"
+	"os"
+	"testing"
+)
+
+func TestGetDockerHost(t *testing.T) {
+
+	tests := []struct {
+		Input    string
+		Expected string
+	}{
+		{
+			"unix:///var/run/docker.sock",
+			"localhost",
+		},
+		{
+			"npipe:////./pipe/docker_engine",
+			"localhost",
+		},
+		{
+			"tcp://1.2.3.4:1234",
+			"1.2.3.4",
+		},
+		{
+			"tcp://1.2.3.4",
+			"1.2.3.4",
+		},
+		{
+			"ssh://1.2.3.4:22",
+			"1.2.3.4",
+		},
+		{
+			"fd://1.2.3.4:1234",
+			"1.2.3.4",
+		},
+		{
+			"",
+			"localhost",
+		},
+		{
+			"invalidValue",
+			"localhost",
+		},
+		{
+			"invalid::value::with::semicolons",
+			"localhost",
+		},
+	}
+	for _, test := range tests {
+		t.Run(fmt.Sprintf("GetDockerHost: %s", test.Input), func(t *testing.T) {
+			// GetHost() uses the DOCKER_HOST environment variable, so we need to set
+			// it to our test input and then reset it afterwards for other tests
+			defer os.Setenv("DOCKER_HOST", os.Getenv("DOCKER_HOST"))
```

Comment:
perhaps it's worth clarifying that setting and resetting the environment variable means we can't safely use `t.Parallel()` here?

I believe it's safe to do this provided we don't call `Parallel()` but somebody with more experience here may well be able to correct me on that.

Reply:
Alternative idea I often use in this case:

1. Create a `getDockerHostFromEnv(env []string) string` method that takes in a list of environment variables (each one in the `key=value` format).
1. Update `GetDockerHost` to call `getDockerHost(os.Environ())` so it's using the real env vars.
1. Update the automated tests to pass in mock values.